lib/mayaUsd/ufe/UsdUndoAttributesCommands.cpp

```diff
@@ -53,12 +53,19 @@ Ufe::Attribute::Ptr UsdAddAttributeCommand::attribute() const
     return UsdAttributes(sceneItem).attribute(_name);
 }
 
+void UsdAddAttributeCommand::setName(const std::string& newName) { _name = newName; }
+
 void UsdAddAttributeCommand::executeUndoBlock()
 {
     // Validation has already been done. Just create the attribute.
     auto sceneItem
         = std::dynamic_pointer_cast<UsdSceneItem>(Ufe::Hierarchy::createItem(_sceneItemPath));
-    UsdAttributes::doAddAttribute(sceneItem, _name, _type);
+    auto attrPtr = UsdAttributes::doAddAttribute(sceneItem, _name, _type);
+
+    // Set the name, since it could have been changed in order to be unique.
+    if (attrPtr) {
+        setName(attrPtr->name());
```

Comment:
Why didn't doAddAttribute() set the name on the return attrPtr?  That way, you wouldn't need the setName(), and you could leave the _name const.

Reply:
Sorry but maybe I didn't understand the comment. 
`doAddAttribute()` actually sets the new name for `attrPtr`. 
The `setName` function sets the name for the `UsdAddAttributeCommand`.

Reply:
Right you are, sorry for the confusion.  Still a bit uncomfortable as the rename command determines the name for a prim in its constructor
https://github.com/Autodesk/maya-usd/blob/158c1d3ea5efa671bbbd6d1a217b60ba02519823/lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp#L92
not at execute time, whereas you are doing the reverse.  There are arguments for both, so I think what you have is fine.

Reply:
That's a really interesting class, I didn't know it, thanks for the info.